spec/clang_spec.lua

```diff
@@ -178,4 +177,154 @@ describe("cursor kind", function()
+                assert.are.equals(cursor_type_spelling, "void (float, float *)")
+        end)
+end)
+
+describe("result type", function()
+        it("succeeds in matching the expected result type", function()
+                local parser = luaclang.newParser("spec/function.c")
+                local cursor = parser:getCursor()
+                cursor = find_decl(cursor)
+                local cursor_type = cursor:getType()
+                assert.has_no.errors(function()
+                        result_type = cursor_type:getResultType()
+                end)
+                local result_type_str = result_type:getTypeSpelling()
+                assert.are.equals(result_type_str, "void")
+        end)
+end)
+
+describe("arg type", function()
+        it("succeeds in matching the expected arg type", function()
+                local parser = luaclang.newParser("spec/function.c")
+                local cursor = parser:getCursor()
+                cursor = find_decl(cursor)
+                local cursor_type = cursor:getType()
+                assert.has_no.errors(function()
+                        arg_type = cursor_type:getArgType(1)
+                end)
+                local arg_type_str = arg_type:getTypeSpelling()
+                assert.are.equals(arg_type_str, "float *")
+        end)
+end)
+
+describe("get num args", function()
+        it("succeeds in getting the number of arguments", function()
+                local parser = luaclang.newParser("spec/function.c")
+                local cursor = parser:getCursor()
+                cursor = find_decl(cursor)
+                local num_args = cursor:getNumArgs()
+                assert.are.equals(num_args, 2)
+        end)
+end)
+
+describe("inline function check", function()
+        it("succeeds in identifying an inline function", function()
+                local parser = luaclang.newParser("spec/function_inline.c")
+                local cursor = parser:getCursor()
+                cursor = find_decl(cursor)
+                assert.are.equals(cursor:isFunctionInlined(), true)
+        end)
+
+        it("succeeds in identifying a non-inline function", function()
+                local parser = luaclang.newParser("spec/function.c")
+                local cursor = parser:getCursor()
+                cursor = find_decl(cursor)
+                assert.are.equals(cursor:isFunctionInlined(), false)
+        end)
+end)
+
+describe("enum constant value", function()
+        it("suceeds in getting the correct integral value of the EnumConstantDecl", function()
+                local parser = luaclang.newParser("spec/enum.c")
+                local cursor = parser:getCursor()  
+                cursor = find_decl(cursor)
+                local last_enumconst
+                cursor:visitChildren(function(cursor, parent)
+                     last_enumconst = parent
+                     return "recurse"
+                end)
+                enum_value = last_enumconst:getEnumValue()
```

Comment:
Test calling this function to a cursor that isn't a enum. Please review all of cases like this.